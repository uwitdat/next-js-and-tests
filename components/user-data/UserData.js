import React, { useState } from 'react'
import { useRouter } from 'next/router';
import AddUserForm from '../add-user-form/AddUserForm';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Image from 'next/image';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import userStyles from './user-styles.module.css';


const UserData = ({ data }) => {
  const [users, setUsers] = useState(data);
  const router = useRouter();

  const handleGoToProfile = (user) => {
    router.push(
      {
        pathname: `/users/${user.id}`,
        query: { user: JSON.stringify(user) },
      },
      `/users/${user.id}`,
    )
  }

  return (
    <section className={userStyles.container}>
      <h1>Users:</h1>
      {users.map((user) => (
        <Card sx={{ maxWidth: 345, width: '70%', margin: '1rem', display: 'flex', flexDirection: 'column', alignItems: 'center' }} key={user.id}>
          <Image
            width={100}
            height={100}
            src={'https://www.kindpng.com/picc/m/24-248253_user-profile-default-image-png-clipart-png-download.png'} />
          <CardContent>
            <Typography gutterBottom variant="h5" component="div">
              {user.name}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              {user.email}
            </Typography>
          </CardContent>
          <CardActions>
            <Button onClick={() => handleGoToProfile(user)} size="small">View Profile</Button>
          </CardActions>
        </Card>
      ))}
      <AddUserForm users={users} setUsers={setUsers} />
    </section>
  )
}

export default UserData

