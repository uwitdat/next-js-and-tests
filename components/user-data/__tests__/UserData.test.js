import React from 'react';
import { render, cleanup, } from "@testing-library/react";
import '@testing-library/jest-dom';
import UserData from '../UserData';
import renderer from 'react-test-renderer';

afterEach(() => {
  cleanup();
});

const users = [
  {
    id: 1,
    name: 'ben',
    email: "email@email.com",
    phone: '111-111-1111',
    website: 'some site'
  },
  {
    id: 2,
    name: 'ben two',
    email: "emailtwo@email.com",
    phone: '222-222-2222',
    website: 'some site two'
  },
];

test('<UserData/> and its props render without crashing', () => {
  const { getByTestId, getByText } = render(<UserData users={users} />);

  users.forEach((user) => {
    const imageComponent = getByTestId(`user-img-${user.id}`);

    expect(getByText(user.name)).toBeVisible();
    expect(getByText(user.email)).toBeVisible();

    expect(imageComponent).toBeInTheDocument();
  });

});


test('<UserData/> matches snapshot', () => {
  const tree = renderer.create(<UserData data={users} />).toJSON();
  expect(tree).toMatchSnapshot();
})