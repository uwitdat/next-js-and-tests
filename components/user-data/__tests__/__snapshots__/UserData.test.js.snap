// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserData/> matches snapshot 1`] = `
<section>
  <h1>
    Users:
  </h1>
  <div>
    <h2>
      Add New User
    </h2>
    <form
      action="#"
      data-testid="add-user-form"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        style={
          Object {
            "margin": ".5rem",
            "width": "100%",
          }
        }
        value=""
      />
      <input
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Name"
        style={
          Object {
            "margin": ".5rem",
            "width": "100%",
          }
        }
        value=""
      />
      <input
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Phone"
        style={
          Object {
            "margin": ".5rem",
            "width": "100%",
          }
        }
        value=""
      />
      <input
        name="website"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Website"
        style={
          Object {
            "margin": ".5rem",
            "width": "100%",
          }
        }
        value=""
      />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
         Submit
      </button>
    </form>
  </div>
</section>
`;
